### PR TITLE
Show Commit Date in About Dialog / Git Describe Format Improvements

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

**Comment by daschuer** [comment]
Thank you. I am just considering, if we should use the compile date in case the commit is modified.
This would help to distinguish two binaries build at home and gets around the reproducible build issue we have, using always the compile date.

File: CMakeLists.txt [hunk 1]
```diff
@@ -333,6 +333,20 @@ else()
...
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  OUTPUT_VARIABLE GIT_COMMIT_DATE
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  ERROR_QUIET
+)
+if(NOT GIT_COMMIT_DATE)
+  message(STATUS "Git commit date: unknown")
+else()
+  message(STATUS "Git commit date: ${GIT_COMMIT_DATE}")
+endif()
```
Comment (daschuer) [inline CMakeLists.txt:348]: Please add a comment at the configure call below.

**Comment by daschuer** [comment]
Not sure if there is a simple solution to show the build time that does not force rebuild it self. So maybe we should scratch the build time idea.

**Comment by Holzhaus** [comment]
> Thank you. I am just considering, if we should use the compile date in case the commit is modified.
> This would help to distinguish two binaries build at home and gets around the reproducible build issue we have, using always the compile date.

I don't understand. Using the compile date will make the build unreproducible in all cases. Note that the git version in the has a `-modified` suffix that helps distinguishing these cases.

**Comment by daschuer** [comment]
All this doe not work reliable, because the project is not reconfigured if a file is changed. This can be fixed in a follow up PR though.

**Comment by daschuer** [comment]
I will take care, Thank you for this PR.

**Pull request merged**